pkgs/tools/system/sleuthkit/default.nix

```diff
@@ -1,9 +1,47 @@
-{ lib, stdenv, fetchFromGitHub, autoreconfHook, libewf, afflib, openssl, zlib }:
+{ lib, stdenv, fetchFromGitHub, autoreconfHook, libewf, afflib, openssl, zlib, openjdk, perl, ant }:
 
 stdenv.mkDerivation rec {
   version = "4.11.0";
   pname = "sleuthkit";
 
+  rdeps = stdenv.mkDerivation rec {
+    # Fetch libraries using a fixed output derivation
+
+    version = "1.0";
+    pname = "sleuthkit-deps";
+    nativeBuildInputs = [ autoreconfHook openjdk ant ];
+
+    src = fetchFromGitHub {
+        owner = "sleuthkit";
+        repo = "sleuthkit";
+        rev = "${pname}-${version}";
+        sha256 = "sha256:1dh06k8grrj3wcx3h9m490p69bw41dz6cv8j5j1drpldmv67k3ki";
+    };
+
+    IVY_HOME = "/build/.ant";
+    phases = "unpackPhase buildPhase installPhase";
```

Comment:
What parts of the default you are trying to avoid here?

Reply:
I needed to split the build into two stages: an inner one that downloads a bunch of jars in a fixed-output derivation, and an outer stage that is, essentially, the previous package build. The inner's jars are needed for building the outer's JNIs.  The phases referenced above are for the inner/downloading stage. The default build mixes building/non-downloading operations with the download and would 'pollute' the fixed output.

In the 'outer' stage, I'm still using the 'default' (with some extra arguments) because by then I have the needed jars and at that point the default build will not re-download what it already has.

Reply:
Yes, but is there a need to specify the inner phases manually?

Reply:
Yes because the default phase list and phase declarations invoke the sleuthkit build process before the jars have been downloaded. Thus I had to override the three phase declarations (unpack build install) shown. I notice that if I leave the phase list itself (phase=..) alone, this build still works, but I think that is unsafe since it invokes configure/autoconf and possibly other steps that the inner stage is not using.  

Here are some options
a. do what I did:  phases = "unpackPhase buildPhase installPhase";
b. leave phases alone and set: configurePhase = "";
c. leave both phases and configurePhase alone and allow configure to do an unused bootstrap/autoconf/configure

Thoughts?

Reply:
If you do not want autoconf, maybe just leave out `autoreconfHook` from `nativeBuildInputs`?

Reply:
I agree, leaving the autoreconfHook in the inner derivation was sloppy and I'll remove it. But configurePhase will still do a unnecessary configure (right?).  My thinking was that writing phases = "unpackPhase buildPhase installPhase" seems clear and avoids the issue that some future sleuthkit revision's 'invisible' configure will break things.

Reply:
On the other hand, it could become necessary! `configurePhase` will run `./configure` if it exists and do nothing otherwise. if you realy want to avoid that, there is `dontConfigure`. I have see some mass-fixing to reduce unnecessary use of `phases`, and `fixupPhase` is often useful in subtle ways, so if the phase list is not immediately necessary, maybe it should be omitted.

Just so that it is said: it looks like the build is special-case enough that supporting `preBuild`/`postBuild` hooks might indeed be unnecessary (for non-fixed-output builds the advice is to invoke `runHook` in the redefined phases, but I don't think it makes sense here).

Reply:
Overwriting phases like this is forbidden. Please use dontPHASE to disable unwanted phases.

Reply:
Given the current state of https://nixos.org/manual/nixpkgs/unstable/#sec-stdenv-phases I would ask for a prooflink to say it is _forbidden_, but indeed here it does not seem to be justified.

Reply:
https://github.com/NixOS/nixpkgs/issues/28910

Reply:
It doesn't look as a consensus for complete forbidding to me. I agree `phases` should not be used here, though.

Reply:
Ok. I've unset phases and used dontConfigure = true;